logbook/logbook.go

```diff
@@ -371,21 +360,72 @@ func (book *Book) WriteDatasetRename(ctx context.Context, ref dsref.Ref, newName
 	if book.listener != nil {
 		book.listener(&Action{
 			Type:       ActionDatasetRename,
-			InitID:     dsLog.ID(),
+			InitID:     initID,
 			PrettyName: newName,
 		})
 	}
 	return book.save(ctx)
 }
 
+// RefToInitID converts a dsref to an initID by iterating the entire logbook looking for a match.
+// TODO(dustmop): Don't depend on this function permanently, use a higher level resolver and
+// convert all callers of this function to use that resolver's initID instead of converting a
+// dsref yet again.
+func (book *Book) RefToInitID(ref dsref.Ref) (string, error) {
+	if book == nil {
+		return "", ErrNoLogbook
+	}
+
+	// RetrieveRef is inefficient, iterates the top two levels of the logbook.
+	// Runs in O(M*N) where M = number of users, N = number of datasets per user.
```

Comment:
It's worse than this. Calculating the `Name()` of a log requires iterating all operations in the log.

Rather than bury a comment about time complexity in the function body, please declare it in the function description.